samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,194 @@
+| Point           |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+| Operation       | Mandatory   | Read, Write |             | Encryption  |
+| Control         |             |             |             | Required    |
+| Point           |             |             |             |             |
+| - Client        |             |             |             |             |
+| Characteristic  |             |             |             |             |
+| Configuration   |             |             |             |             |
+| descriptor      |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+| Data Out        | Mandatory   | Notify      |             | Encryption  |
+|                 |             |             |             | Required    |
++-----------------+-------------+-------------+-------------+-------------+
+| Data Out        | Mandatory   | Read, Write |             | Encryption  |
+| - Client        |             |             |             | Required    |
+| Characteristic  |             |             |             |             |
+| Configuration   |             |             |             |             |
+| descriptor      |             |             |             |             |
++-----------------+-------------+-------------+-------------+-------------+
+
+The purpose of each characteristic is as follows:
+
+  * ``Information``: for client to get info message from server.
+  * ``Operation Control Point``: for client to send request message to server, and server to send response message to client.
+  * ``Data Out``: for server to send result message to the client.
+
+Advertisement
+-------------
+
+After powerup, the device checks if it is provisioned. If it is, it loads saved credentials and tries to connect to the specified access point.
+After 30 seconds, the device advertises over Bluetooth LE. The operation times out if the attempt fails.
+If the device is not provisioned, it advertises immediately.
+
+The advertising data contains several fields to facilitate provisioning.
+It contains the UUID of the Wi-Fi Provisioning Service and four bytes of service data, as described in the following table.
+
++----------+----------+----------+----------+
+|  Byte 1  |  Byte 2  |  Byte 3  |  Byte 4  |
++==========+==========+==========+==========+
+| Version  | Flag(LSB)| Flag(MSB)|   RSSI   |
++----------+----------+----------+----------+
+
+* Version: 8-bit unsigned integer. The value is the version of the Wi-Fi Provisioning Service sample.
+
+* Flag: 16-bit little endian field. Byte 2 (first byte on air) is LSB, and Byte 3 (second byte on air) is MSB.
+
+  - Bit 0: provisioning status. 1 if device is provisioned, 0 otherwise.
+
+  - Bit 1: connection status. 1 if Wi-Fi is connected, 0 otherwise.
+
+  - Bit 2 - 15: RFU
```

Comment:
```suggestion
  * Bit 0: provisioning status. 1 if device is provisioned, 0 otherwise.

  * Bit 1: connection status. 1 if Wi-Fi is connected, 0 otherwise.

  * Bit 2 - 15: RFU
```